src/server/api/controllers/vessel.controller.js

```diff
@@ -0,0 +1,118 @@
+"use strict";
+
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+
+//GET
+const getVessels = async req => {
+  try {
+    return await knex("vessels").select(
+      "vessels.id",
+      "vessels.organization_id",
+      "vessels.mmsi",
+      "vessels.name"
+    );
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+//GET:id
+const getVesselById = async id => {
+  try {
+    const vessels = await knex("vessels")
+      .select("vessels.id as id", "organization_id", "mmsi", "name")
+      .where({ id: id });
+    if (vessels.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `Cannot find vessel for ID ${id}!`,
+        404
+      );
+    }
+    return vessels;
+  } catch (err) {
+    return err.message;
+  }
+};
+//POST
+
+const createVessel = async ({ body }) => {
+  const { organization_id, mmsi, name } = body;
+
+  const organization = await knex
+    .from("organizations")
+    .select("*")
+    .where({
+      id: organization_id
+    });
+
+  if (organization.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `${organization_id} organization not found!`,
+      404
+    );
+  }
+
+  const vessels = await knex("vessels").insert({
+    organization_id: organization[0].id,
+    mmsi: mmsi,
+    name: name
+  });
+  return vessels;
+};
+
+//PATCH
+const editVessel = async ({ body }) => {
+  const { id, organization_id, mmsi, name } = body;
+  const vessel = await knex
```

Comment:
Also to make it more clear that you are expecting a single result from the query you should be able to do the following `knex('vessels').where({name}).first()` (using `first()`) the way I see it from the knex docs